src/ecs/rendering/components/texture.rs

```diff
@@ -187,7 +187,16 @@ impl Context for TextureContext {
         if image.w > u16::max_value() as usize || image.h > u16::max_value() as usize {
             return TextureFuture::unsupported_size(image.w, image.h);
         }
-
+        
+        // Flip the image horizontally for OpenGL, when other backends are added
+        // DO NOT execute this for statement on them.
```

Comment:
We have the cfg already, so why not use it right now?

Reply:
It's present in amethyst_renderer but not in amethyst.

Reply:
Hm.. I would still be in favor of doing it right now. Maybe we could just add the flag to the main crate?

Reply:
I agree, although can we do that in a follow up pull request? I'd like to also add the necessary shaders so we can turn on all of our backends.

Reply:
I'm not sure you need to do such low-level rendering related stuff in main crate.
'amethyst_renderer/src/tex.rs' is better place for do such hacks 😄

Reply:
@omni-viral The patch requires the ability to mutate the underlying buffer, and I'd prefer to not mutate a buffer unless we're actively consuming it.  `TextureBuilder::new` doesn't currently consume the data.  Should it be made to?

Reply:
Alternatively: Make it just take a mutable reference, flip it, send it to the GPU, flip it back, release the borrow.

Side note: This is getting more hacky by the second.

Reply:
@Xaeroxe you can flip it in `TextureBuilder::build`

Reply:
Each row should be flipped or rows order?

Reply:
@omni-viral just the row order.  Thanks for the idea, I'll do it in `build`

Reply:
nit: flipping along the y-axis is called 'vertical flip' (I always mess this up always)

Reply:
Oh, thanks @msiglreith